src/strategies/strategies/arrow-vesting/index.ts

```diff
@@ -38,59 +38,154 @@ export async function strategy(
     blockTag: blockTag
   });
 
-  const vestingFactoryMulti = new Multicaller(
-    network,
-    provider,
-    vestingFactoryAbi,
-    { blockTag }
-  );
-
-  [...Array(vestingContractCount.toNumber()).keys()].forEach(contractIdx => {
-    vestingFactoryMulti.call(contractIdx, options.vestingFactory, 'escrows', [
-      contractIdx
-    ]);
-  });
-
-  const vestingContracts: Record<number, string> =
-    await vestingFactoryMulti.execute();
-
-  // Get all vesting contract parameters.
-  const vestingContractMulti = new Multicaller(
-    network,
-    provider,
-    vestingContractAbi,
-    { blockTag }
-  );
-
-  Object.values(vestingContracts).forEach(vestingContractAddress => {
-    vestingContractMulti.call(
-      `${vestingContractAddress}.recipient`,
-      vestingContractAddress,
-      'recipient',
-      []
-    );
-    vestingContractMulti.call(
-      `${vestingContractAddress}.total_locked`,
-      vestingContractAddress,
-      'total_locked',
-      []
-    );
-    vestingContractMulti.call(
-      `${vestingContractAddress}.start_time`,
-      vestingContractAddress,
-      'start_time',
-      []
-    );
-    vestingContractMulti.call(
-      `${vestingContractAddress}.unclaimed`,
-      vestingContractAddress,
-      'unclaimed',
-      []
+  const totalContracts = vestingContractCount.toNumber();
+  const batchSize = 30; // Process contracts in batches of 20 to avoid RPC timeouts
+  const maxContracts = 200; // Limit total contracts to prevent excessive processing time
```

Comment:
Will this not miss tokens locked up in contracts beyond the `maxContracts` limit and return incorrect totals for some users?